### PR TITLE
Refactor continued

### PR DESCRIPTION
**Pull request by apaschos**
The real diff is in the last commit. What changed is I added a class and moved the big functions inside as members functions. Filtering out the whitespace should make it easy to see the changes.

### CODE REVIEW

File: gnat2goto/ireps/ireps_generator.py [hunk 1]
```diff
@@ -205,2067 +205,2071 @@ def mk_prefixed_lines(prefix, lines, join=""):
...
+        schemata = {}
+        for schema_fn in schema_file_names:
+            with open(schema_fn, "rU") as fd:
+                sn = os.path.splitext(os.path.basename(schema_fn))[0]
+                # print "Loading %s" % sn
+                schemata[sn] = json.load(fd)
+                schemata[sn]["subclasses"] = set()
+                schemata[sn]["ada_name"] = ada_casing("i_" + sn)
+
+        # schemata will contain the following. Things in <> we synthesise below.
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/ireps/ireps_generator.py]: Why was this changed from upper to lower case?

File: gnat2goto/ireps/ireps_generator.py [hunk 2]
```diff
@@ -205,2067 +205,2071 @@ def mk_prefixed_lines(prefix, lines, join=""):
...
+                            op_counts[sn]["int"] += 1
+                        else:
+                            print sn, setter_name, kind, typ
+                            assert False
+
+        MAX_INTS  = max(x["int"] for x in op_counts.itervalues())
+        MAX_BOOLS = max(x["bool"] for x in op_counts.itervalues())
+
+        if optimize:
+            layout, MAX_INTS, MAX_BOOLS = self.optimize_layout(layout,
```
Comment (hannes-steffenhagen-diffblue) [inline gnat2goto/ireps/ireps_generator.py]: This seems to have been broken in the previous version? It looks like this used to be an empty `if`, which isn't actually valid in Python.

File: gnat2goto/ireps/ireps_generator.py (hunk 1 above)
Comment (apaschos) [reply gnat2goto/ireps/ireps_generator.py]: Accidentally. Will be fixed soon.

File: gnat2goto/ireps/ireps_generator.py (hunk 2 above)
Comment (apaschos) [reply gnat2goto/ireps/ireps_generator.py]: I just checked it with meld locally. It seems it's a problem with github diff tool or maybe the "ignore whitespace" option. It was working before, it is also working now.
